src/content/docs/en/guides/integrations-guide/vercel.mdx

```diff
@@ -333,11 +333,11 @@ You can use Vercel Edge middleware to intercept a request and redirect before se
 **Trying to rewrite?** Currently rewriting a request with middleware only works for static files.
 :::
 
-### Vercel Edge Middleware with Astro middleware
+### Running Astro middleware on Vercel Edge Functions
 
-The `@astrojs/vercel/serverless` adapter can automatically create the Vercel Edge middleware from an Astro middleware in your code base.
+The `@astrojs/vercel/serverless` adapter can create an [edge function](https://vercel.com/docs/functions/edge-functions) from an Astro middleware in your code base. When `edgeMiddleware` is enabled, an edge function will execute your middleware code for all requests for static assets, prerendered pages, and on-demand rendered pages. For on-demand rendered pages, the locals are serialized using JSON and sent in a header for the serverless function, which performs the rendering. As a security measure, the serverless function will refuse to serve requests with a `403 Forbidden` response unless they come from the generated edge function.
```

Comment:
```suggestion
The `@astrojs/vercel/serverless` adapter can create an [edge function](https://vercel.com/docs/functions/edge-functions) from an Astro middleware in your code base. When `edgeMiddleware` is enabled, an edge function will execute your middleware code for all requests for static assets, prerendered pages, and on-demand rendered pages.

For on-demand rendered pages, the locals are serialized using JSON and sent in a header for the serverless function, which performs the rendering. As a security measure, the serverless function will refuse to serve requests with a `403 Forbidden` response unless they come from the generated edge function.
```
Broke this into 2 paragraphs again. If you decide to change the similar Netlify text, make the same change to it here, too!